### ISSUE TITLE
mean/sum/stddev of multiple dimensions

### ISSUE THREAD

**Issue by dsyme** [DiffSharp/DiffSharp#216]
We should add overloads of mean, sum, stddev etc that take multiple dimensions.

From the torch documentation for [mean](https://pytorch.org/docs/stable/generated/torch.mean.html):

> "Returns the mean value of each row of the input tensor in the given dimension dim. If dim is a list of dimensions, reduce over all of them"

**Comment by dsyme** [comment]
Here's the code if you need this before it's added:

```fsharp
type Tensor with 
    member t.mean (dims: int[], ?keepDim: bool) =
           (t, Array.rev (Array.sort dims)) ||> Array.fold (fun input dim -> dsharp.mean(input, dim, ?keepDim=keepDim))

    member t.variance (dims: int[], ?keepDim: bool) =
           (t, Array.rev (Array.sort dims)) ||> Array.fold (fun input dim -> dsharp.variance(input, dim, ?keepDim=keepDim))

    member t.sum (dims: int[], ?keepDim: bool) =
           (t, Array.rev (Array.sort dims)) ||> Array.fold (fun input dim -> dsharp.sum(input, dim, ?keepDim=keepDim))

    member t.stddev (dims: int[], ?keepDim: bool) =
           (t, Array.rev (Array.sort dims)) ||> Array.fold (fun input dim -> dsharp.stddev(input, dim, ?keepDim=keepDim))

    member t.moments () =
           dsharp.mean(t), dsharp.stddev(t)

    member t.moments (dim: int, ?keepDim: bool) =
           t.mean(dim, ?keepDim=keepDim), t.stddev(dim, ?keepDim=keepDim)

    member t.moments (dims: int[], ?keepDim: bool) =
           t.mean(dims, ?keepDim=keepDim), t.stddev(dims, ?keepDim=keepDim)


type dsharp with 
    static member mean (input: Tensor, dims: int[], ?keepDim: bool) = input.mean(dims, ?keepDim=keepDim)

    static member variance (input: Tensor, dims: int[], ?keepDim: bool) = input.variance(dims, ?keepDim=keepDim)

    static member sum (input: Tensor, dims: int[], ?keepDim: bool) = input.sum(dims, ?keepDim=keepDim)

    static member stddev (input: Tensor, dims: int[], ?keepDim: bool) = input.stddev(dims, ?keepDim=keepDim)

    static member moments (input: Tensor) = input.moments()

    static member moments (input: Tensor, dim: int, ?keepDim: bool) = input.moments(dim, ?keepDim=keepDim)

    static member moments (input: Tensor, dims: int[], ?keepDim: bool) = input.moments(dims, ?keepDim=keepDim)
```

**Comment by dsyme** [comment]
Tracking this in #227
